### PR TITLE
Phaser renderer misc fixes

### PR DESCRIPTION
**Pull request by pavle-goloskokovic**
Fixes all observed rendering discrepancies listed here: https://github.com/moddio/taro/issues/306

### CODE REVIEW

**Review by Grenagar — APPROVED** [review]

File: ts/src/renderer/phaser/classes/PhaserUnit.ts [hunk 1]
```diff
@@ -37,7 +37,13 @@ class PhaserUnit extends Phaser.GameObjects.Container {
 
 		this.followListener = unit.on('follow', () => {
 			console.log('PhaserUnit follow', unit.id()); // TODO remove
-			scene.cameras.main.startFollow(this, true, 0.05, 0.05);
+			const camera = scene.cameras.main as Phaser.Cameras.Scene2D.Camera & {
```
Comment (sloont) [inline ts/src/renderer/phaser/classes/PhaserUnit.ts:40]: could you explain the syntax here? I obviously trust you with this, I am just confused

File: ts/types/IgeEngine.d.ts [hunk 2]
```diff
@@ -4,11 +4,11 @@ declare class IgeEngine extends IgeClass {
...
 
-    isClient: boolean;
-    isServer: boolean;
+	isClient: boolean;
+	isServer: boolean;
 
-    client: Client;
-    server: Client;
+	client: Client;
+	server: Client;
```
Comment (sloont) [inline ts/types/IgeEngine.d.ts:11]: I'm hung up on this. What case(s) is there `ige.server` that exists and is of type `Client`?

**Review by sloont — APPROVED** [review]

File: ts/src/renderer/phaser/classes/PhaserUnit.ts (hunk 1 above)
Comment (pavle-goloskokovic) [reply ts/src/renderer/phaser/classes/PhaserUnit.ts:40]: I'm extending camera type with `_follow` which is private, but I need to use it to determine if camera is already following that entity. compiled code is much simpler:
https://github.com/moddio/taro/pull/324/files/26cdb18fb3e2dfb0a9681e5b970ab99120750654#diff-25cecd88e11153f0ecdf869e5e323ce73301c622fa4ebff0241dced268cc175aR31-R34

File: ts/types/IgeEngine.d.ts (hunk 2 above)
Comment (pavle-goloskokovic) [reply ts/types/IgeEngine.d.ts:11]: `ige.server` is always assigned here: https://github.com/moddio/taro/blob/master/src/index.js#L11
notice that `App` is not a type but rather name of the argument passed to `init` method.
and only place where it is instantiated is here: https://github.com/moddio/taro/blob/master/src/index.js#L19
and is passed `Client` type. So, yes, `ige.server` is always of type `Client`. Very unfortunate naming, I agree.

**Pull request merged**